### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by JEFFKAY16**
# OOP school library: set up associations

In this update, I implemented the following instructions:
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = rentals
```
Comment (MrBrN197) [inline book.rb]: - [x] Hi, please notice that you are attempting to use a variable named `rentals` which does not exist. therefore your variable `@rentals` will be set to `nil`. you should simply set rentals to and empty array.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  def add_rentals(person, date)
```
Comment (MrBrN197) [inline book.rb:13]: - [x] [OPTIONAL] in terms of naming this function should not be pluralized. the method name should always define what the method does. this method is only adding *one* rental.
```suggestion
  def add_rental(person, date)
```

File: person.rb [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
+    of_age? || @parent_permission ? true : false
+  end
+
+  def validate_name
+    @name = @corrector.correct_name(@name)
   end
 
-  def correct_name
-    @name
+  def add_rentals(book, date)
```
Comment (MrBrN197) [inline person.rb:25]: - [x] [OPTIONAL] in terms of naming this function should not be pluralized. the method name should always define what the method does. this method is only adding *one* rental.
```suggestion
  def add_rental(book, date)
```

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @JEFFKAY16  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ All linter checks are passing correctly
- ✔ code is organized well.

## Required Changes ♻️
_Check the comments under the review._

- [x] Hi, In your `Student` class you have the variable `classroom` which doesn't exist. Please correct this to `_classroom` which is the parameter that is passed into the `initalize` method of this class.

- *Check below on the inline comments for some suggested changes.*

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (JEFFKAY16) [reply book.rb]: Done

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @JEFFKAY16, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
 
 - [ ] As the previous reviewer suggested, In your `student.rb` file you have used the variable `classroom` which doesn't exist. Please correct this to `_classroom` which is the parameter that is passed into the initialize method of this class. I have added the screenshot as well.
 
 Screenshot from 2022-04-07 23-40-24

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

_**As previous reviewer suggested.**_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**